### PR TITLE
ISPN-10090 Allow running TestNG tests without TestNGTestListener

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-10090

* Don't check for leaks in AbstractInfinispanTest
* Change listeners to crash if they detect a leak
  (and TestNGSuiteChecksTest didn't detect it first)
* Include TestNGSuiteChecksTest in default profile in core
* Log the file name when dumping threads programmatically (ISPN-10024)

### CODE REVIEW

File: commons-test/src/main/java/org/infinispan/commons/test/TestNGTestListener.java [hunk 1]
```diff
@@ -82,6 +84,7 @@ public void onStart(ISuite suite) {
 
    @Override
    public void onFinish(ISuite suite) {
+      ThreadLeakChecker.checkForLeaks(null);
```
Comment (karesti) [inline commons-test/src/main/java/org/infinispan/commons/test/TestNGTestListener.java]: I know this is testing code, but what about creating another test called like checkForLeaksOnFinish() or something like that, instead of passing null and doing some logic inside the checkForLeaks?

Comment (danberindei) [reply commons-test/src/main/java/org/infinispan/commons/test/TestNGTestListener.java]: I looked into this and it turns out the `testName` parameter isn't needed at all -- it was only necessary before to call `testFinished(testName)`, and we don't do that any more.

Comment (karesti) [reply commons-test/src/main/java/org/infinispan/commons/test/TestNGTestListener.java]: good ! 💃
